##Purchase Frequency Analysis

**Objective:**
Calculate the Purchase Frequency using the cleaned e-commerce dataset.

In [ ]:
import pandas as pd

# Load cleaned dataset
df = pd.read_excel("cleaned_online_retail.xlsx")

# Calculate metrics
total_orders = df["InvoiceNo"].nunique()
total_customers = df["CustomerID"].nunique()

purchase_frequency = total_orders / total_customers

# Create summary table
results = pd.DataFrame({
    "Metric": ["Total Orders", "Total Customers", "Purchase Frequency"],
    "Value": [total_orders, total_customers, round(purchase_frequency, 2)]
})

print("\nPURCHASE FREQUENCY ANALYSIS RESULTS")
print("-" * 40)
print(results.to_string(index=False))


PURCHASE FREQUENCY ANALYSIS RESULTS
----------------------------------------
            Metric    Value
      Total Orders 15434.00
   Total Customers  3263.00
Purchase Frequency     4.73


**Conclusion**: Purchase Frequency was calculated using the cleaned e-commerce dataset by dividing the total number of unique orders by the total number of unique customers. The analysis showed a Purchase Frequency of 4.73, indicating that customers placed an average of 4.73 orders during the observed period. This metric will be used in Customer Lifetime Value (CLTV) calculations.